### PR TITLE
quick refactor for item metadata extensibility

### PR DESCRIPTION
**Pull request by ryanfkeepers**
#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #2708

#### Test Plan

- [x] :zap: Unit test

### CODE REVIEW

File: src/pkg/backup/backup.go [hunk 1]
```diff
@@ -49,9 +49,6 @@ type Backup struct {
 	// stats are embedded so that the values appear as top-level properties
 	stats.ReadWrites
 	stats.StartAndEndTime
-
-	// Deprecated
-	stats.Errs // replaced with fault.Errors
```
Comment (vkamra) [inline src/pkg/backup/backup.go:54]: I'm assuming this doesn't cause unmarshal of older serialized backup models to fail. I think the default behavior is that missing fields are just ignored

File: src/internal/connector/graph/errors.go [hunk 2]
```diff
@@ -313,3 +314,32 @@ func appendIf(a []any, k string, v *string) []any {
 
 	return append(a, k, *v)
 }
+
+// MalwareInfo gathers potentially useful information about a malware infected
+// drive item, and aggregates that data into a map.
+func MalwareInfo(item models.DriveItemable) map[string]any {
```
Comment (vkamra) [inline src/internal/connector/graph/errors.go:320]: Minor - use const for `created_by` and other fields. Add a simple unit test to validate this method

**Review by vkamra — APPROVED** [review]

File: src/pkg/backup/backup.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/pkg/backup/backup.go:54]: Yup, shouldn't cause any serialization errors.  Worst case is that old backups lose the count on `Completed (X errors)`

File: src/internal/connector/graph/errors.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/connector/graph/errors.go:321]: done

**Pull request merged**